pkg/utils/helper.go

```diff
@@ -59,20 +64,29 @@ func RunCommandOnHost(command string, arg ...string) (string, error) {
 	return string(out), nil
 }
 
-// RunCommandOnContainer runs a command on container system
-func RunCommandOnContainer(command string, arg ...string) (string, error) {
+// RunCommandOnContainerWithOutputStreams runs a command on container system and returns both the stdout and stderr output streams
+func RunCommandOnContainerWithOutputStreams(command string, arg ...string) (CommandOutputStreams, error) {

```

Comment:
💡I like what you have done, thankyou. To more fully share some thoughts, this way we need to make sure all exisiting functionality are working fine, otherwise, if you find any issue latter here is my 2 cents of idea to un-block you in those scenarios: Why not create separate helper function for the specific OSM needs? Will be a good use of extending Function for the generic function of OSM/SMi specific need?

We could do this as more systematically i.e..:

* Create new helper function with this name and keep old one as it is so that old code consumes whatever it is soncusming right now.
* Once OSM\SMI code is consuming the new code then transition old soncumers into new generic function?

Feel free to add thoughts once you have tested this,

Reply:
_Create new helper function with this name and keep old one as it is so that old code consumes whatever it is soncusming right now._ isn't this exactly what has been done? (see line 87 of helper.go). Unit test would be nice but this change seems straightforward enough that it doesn't require it

Reply:
💡  key point I was trying to bring across as idea share is to be watchful of the the old function - now consumes new function with few cabbages, hence as part of this change the it could be litte onerous to check all impact. 

Hence, if that turns out to be hard to test in any way possible feel free to add this only for use of new functionality and carefully port old functionality to use new function. Thanks.

Reply:
💡 @shailer, this comment was to be careful to make sure all current functionality will get affected in what extent? (So that this could be documented) because the new way the old method signature is now: (Also, I testing will be key for this as well) 

* `RunCommandOnContainer` now consumes `RunCommandOnContainerWithOutputStreams` and new method introduced now use `commandStream`? 

If we have added\changed\introduced new log behaviours it will be key to capture those for documentation purposes please.

``` 
// RunCommandOnContainer  runs a command on container system and returns the stdout output stream
 func RunCommandOnContainer(command string, arg ...string) (string, error) {
 	outputStreams, err := RunCommandOnContainerWithOutputStreams(command, arg...)
 	return outputStreams.Stdout, err
 }
```

Reply:
@Tatsinnit this particular change is really quite trivial and I don't think it needs any new documentation or further work in this PR...

CommandOutputStreams (which I think is what you mean by commandStream), is simply a new struct wrapping both stdOut and stdErr, defined a couple lines up in this file (helpers.go lines 14-18).

The current functionality is maintained _exactly_ by having RunCommandOnContainer only return the stdOut field of this struct.

Reply:
@davidkydd, yes, I am aware. Like I have mentioned few times to you, my comments when it comes down to exploring existing functionality are purely careful and defensive, it is upto author to take the thoughts and decide the design flow, the ideas I have given it purely be cautious and aware in case anything new functionality will start appearing for existing feature.

* Basically, keep in mind and if we are adding any new thing to an existing functionality I want it to be added as simple as one liner, either as comment, for anyone else to be aware and it will help other readers.

Reply:
I agree that they are fair comments, and show admirable intent!

Are you satisfied with @shalier's explanation of the local testing that has been performed? In which case we can look to resolve this particular thread?

Reply:
@davidkydd @Tatsinnit  SGTM to resolve this thread, thanks for the explanation!